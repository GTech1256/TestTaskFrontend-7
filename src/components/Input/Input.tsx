import React, { useCallback, FunctionComponent } from 'react'


export type Props = {
  className?: string
  value?: string
  onChange: (value: string) => void
  placeholder?: string
  label?: string
  name: string
}

const Input: FunctionComponent<Props> = ({
  className = '',
  onChange,
  value,
  placeholder,
  label,
  name
}) => {
  const handleChange = useCallback(({ target: { value } }) => {
    onChange(value)
  }, [onChange])


  return (
    <div className={`input-group mb-3 ${className}`}>
      {label && (
        <div className="input-group-prepend">
          <label className="input-group-text" htmlFor={name}>{label}</label>
        </div>
      )}
      <input
        className="form-control"
        name={name}
        placeholder={placeholder}
        aria-label={name}
        aria-describedby="basic-addon1"
        value={value}
        onChange={handleChange}
      />
    </div>
  )
}

export default Input