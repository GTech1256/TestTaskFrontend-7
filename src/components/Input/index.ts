
export type { Props as InputProps } from './Input'
export { default } from './Input'