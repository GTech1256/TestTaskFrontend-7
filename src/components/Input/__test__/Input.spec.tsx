import React from 'react';
import renderer from 'react-test-renderer';

import Input, { InputProps } from '../index'


const props: InputProps = {
  className: 'test-class-name',
  name: 'testSelect',
  label: 'testTitle',
  onChange: () => {}
}

describe('<Input />', () => {
  it('should render correct', () => {
    const tree = renderer
      .create(<Input {...props} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  })
})