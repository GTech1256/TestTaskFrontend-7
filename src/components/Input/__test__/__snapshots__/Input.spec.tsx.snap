// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render correct 1`] = `
<div
  className="input-group mb-3 test-class-name"
>
  <div
    className="input-group-prepend"
  >
    <label
      className="input-group-text"
      htmlFor="testSelect"
    >
      testTitle
    </label>
  </div>
  <input
    aria-describedby="basic-addon1"
    aria-label="testSelect"
    className="form-control"
    name="testSelect"
    onChange={[Function]}
  />
</div>
`;
